api/consul.go

```diff
@@ -369,9 +372,18 @@ func (tc *ConsulGatewayTLSConfig) Copy() *ConsulGatewayTLSConfig {
 		return nil
 	}
 
-	return &ConsulGatewayTLSConfig{
-		Enabled: tc.Enabled,
+	result := &ConsulGatewayTLSConfig{
+		Enabled:       tc.Enabled,
+		TLSMinVersion: tc.TLSMinVersion,
+		TLSMaxVersion: tc.TLSMaxVersion,
 	}
+	if len(tc.CipherSuites) != 0 {
+		cipherSuites := make([]string, len(tc.CipherSuites))
+		copy(cipherSuites, tc.CipherSuites)
+		result.CipherSuites = cipherSuites
+	}
```

Comment:
Would you mind writing some test for this? `Copy`methods have caused some issues for us in the past 😅